src/components/Home.jsx

```diff
@@ -2,50 +2,41 @@ import React, { useState, useRef, useEffect } from 'react';
 import CharacterList from './characterList';
 
 const Home = ({ setActors, actors, currentScript }) => {
-  // const [actors, setActors] = useState([]);
-
-  const actorFirstName = useRef('');
-  const actorLastName = useRef('');
+  const newActorInput = useRef('');
 
   // submit of actor name handler
-  const handleInput = (event) => {
-    let newActors = [...actors];
-    let actor = {
-      firstName: actorFirstName.current.value,
-      lastName: actorLastName.current.value,
-    };
-    newActors.push(actor);
-    // set actors in state to be our new actors
-    setActors(newActors);
-    // reset input value to empty string to clear out old input
-    actorFirstName.current.value = '';
-    actorLastName.current.value = '';
+  const newActorHandler = (event) => {
+    event.preventDefault();
+    const newActor = {name: newActorInput.current.value};
 
-    // fetch request to the backend to add the new actor to the database
+    // Add the new actor to the database
     fetch('/actors', {
       method: 'POST',
       headers: {
         'Content-Type': 'application/json',
       },
-      body: JSON.stringify(actor),
+      body: JSON.stringify(newActor),
+    }).then(() => {
+      // set actors in state to be our new actors
+      setActors([...actors, newActor]);
+      // reset input value to empty string to clear out old input
+      newActorInput.current.value = '';
```

Comment:
these were moved to _after_ the request completes so that if it fails the ui won't look like the actor was added.